### PR TITLE
Create contest

### PR DESCRIPTION
**Pull request by mctran-phi**
### What this PR does (required):
- Implemented onClick to select images
- Refactored useState into using Formik
- HandleSubmit to post contest information to MongoDB

### Screenshots / Videos (required):
- Screen Shot 2021-07-20 at 1 24 51 PM

### Any information needed to test this feature (required):
- Go to create-contest route and fill in all the fields. Click create contest button and it will make a post request to the server to store the data in  MongoDB.

### Any issues with the current functionality (optional):
- Date is incorrect from changing moment to date. May have to format moment to a string and replace Date type from Mongo Schema to string

### CODE REVIEW

File: client/src/pages/CreateContest/CreateContestForm/CreateContestForm.tsx [hunk 1]
```diff
@@ -1,144 +1,228 @@
...
-    </Box>
+          <Box mt={3} mb={3} className={classes.box}>
+            <Typography className={classes.label}>Which designs do you like?</Typography>
+            <Typography className={classes.sub} component='p'>
+              {'Let\'s start by helping your designers understand which styles your prefer.'}
+            </Typography>
+            <Box mt={3} mb={3} className={classes.imageList}>
+              <ImageList rowHeight={160} cols={4} className={classes.images}>
+                {dogImages.map((image: string, id: number) => (
+                  <ImageListItem key={id}>
```
Comment (rajivtitus) [inline client/src/pages/CreateContest/CreateContestForm/CreateContestForm.tsx]: Avoid using indexes as keys. Can we use something unique here?

Ref: https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

File: client/src/pages/CreateContest/CreateContest.tsx [hunk 2]
```diff
@@ -3,11 +3,41 @@ import Box from '@material-ui/core/Box';
...
+  ) => {
+    const contest = {
+      title: title,
+      description: description,
+      prizeAmount: prizeAmount,
+      deadlineDate: deadlineDate.toDate(),
+      images: images,
+    };
+
+    axios.post('/contest', contest)
```
Comment (rajivtitus) [inline client/src/pages/CreateContest/CreateContest.tsx]: Can we please create a helper function here to stay consistent with the code base?

**Review by rajivtitus — COMMENTED** [review]
Looks good! Just a few comments, please have a look. Thanks

**Comment by mctran-phi** [comment]
I made the changes, please take a look!

File: client/src/pages/CreateContest/CreateContestForm/CreateContestForm.tsx [hunk 3]
```diff
@@ -81,15 +76,15 @@ export default function CreateContestForm({ handleSubmit }: Props): JSX.Element
...
-      let newDate = moment();
-      if (zone === 'PDT') {
-        newDate = contest.deadlineDate.tz("America/Los_Angeles", true);
-      } else if (zone === 'CDT') {
-        newDate = contest.deadlineDate.tz("America/Chicago", true);
-      } else {
-        newDate = contest.deadlineDate.tz("America/New_York", true);
-      }
-      handleSubmit({ ...contest, deadlineDate: newDate, images: images });
+      // let newDate = moment();
```
Comment (rajivtitus) [inline client/src/pages/CreateContest/CreateContestForm/CreateContestForm.tsx:79]: I presume you will be using this code in the future? If not you can exclude it.

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes.

**Pull request merged**